fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeStmt.java

```diff
@@ -60,47 +60,50 @@
  *      properties: properties of statistics jobs
  */
 public class AnalyzeStmt extends DdlStmt {
-    private static final Logger LOG = LogManager.getLogger(AnalyzeStmt.class);
-
-    // time to wait for collect  statistics
+    /** time to wait for collect  statistics */
     public static final String CBO_STATISTICS_TASK_TIMEOUT_SEC = "cbo_statistics_task_timeout_sec";
 
     private static final ImmutableSet<String> PROPERTIES_SET = new ImmutableSet.Builder<String>()
             .add(CBO_STATISTICS_TASK_TIMEOUT_SEC)
             .build();
 
-    public static final Predicate<Long> DESIRED_TASK_TIMEOUT_SEC = (v) -> v > 0L;
+    private static final Predicate<Long> DESIRED_TASK_TIMEOUT_SEC = (v) -> v > 0L;
 
     private final TableName dbTableName;
+    private final PartitionNames partitionNames;
     private final List<String> columnNames;
     private final Map<String, String> properties;
 
     // after analyzed
     private long dbId;
     private final Set<Long> tblIds = Sets.newHashSet();
 
-    public AnalyzeStmt(TableName dbTableName, List<String> columns, Map<String, String> properties) {
+    public AnalyzeStmt(TableName dbTableName,
+            List<String> columns,
+            PartitionNames partitionNames,
```

Comment:
It is impossible to change only the constructor without changing the semantic parsing. 
These two need to be modified together.